### DISCUSSION TITLE
setup-wsl + actions/checkout  ?

### DISCUSSION THREAD

**Discussion by MarximusMaximus** [Vampire/setup-wsl#32] [Q&A] [answered]
I'm not sure if this issue is for setup-wsl or actions/checkout.

I've run setup-wsl, and that worked. But I'm trying to use actions/checkout now to checkout the repo and it fails in "neat" ways.

1) it will default to using windows git and muck up all line endings
1.a)
 - add `git` and `git-lfs` additional packages to WSL
 - write a `git.cmd` override that is `wsl git %*` so that we use WSL's git
 - setup WSLENV so that GITHUB_* and RUNNER_* env vars get passed down to WSL
 - basic git commands now work and use WSL's git!
2) actions/checkout does indeed use wsl's git, but uses all windows paths (presumably b/c it's running via node in win32)
2.a) have `git.cmd` use `${{ steps.setup-wsl.outputs.wsl-shell-wrapper-path }} git %*` instead, but that just yields: 
```
Invalid arguments 
Usage: wsl-sh [-u <user>] <script file a.k.a. {0}> 
```
2.b) tried to override node in the same way as git, but it doesn't work (it either still uses the WIN32 node, not WSL node; or is using the WIN32 env vars and not the WSL env vars)

Any ideas would be greatly appreciated.

**Comment by Vampire** [comment, marked as answer]
Hi there, I converted the issue to a discussion as I think this is more a usage question, not an issue. We can reopen the issue if I'm wrong.

Why do you actually try to use Git from inside WSL? Just use the Windows Git and use the checked out files in WSL.

If by mucked up line endings you mean that shell scripts have Windows line endings, then you should add a `.gitattributes` file that fixes this. See #20 for more information.

If that is not enough in your opinion, can you provide an example of what you mean / what you try, and why?

**Comment by MarximusMaximus** [comment]
Hmm. I'll look into that and get back to you. Thank you for the advice.

**Comment by MarximusMaximus** [comment]
Confirmed. Closing as this is solvable via .gitattributes (explicitly setting files end of line characters), either:
```
*                   text=auto eol=lf

# Windows Shell Script (MUST be crlf)
*.bat               text eol=crlf
```
or like:
```
# Linuxy Shell Script (MUST be lf)
*.sh                text eol=lf

# Windows Shell Script (MUST be crlf)
*.bat               text eol=crlf
```
